src/uk/co/ben_gibson/repositorymapper/Host/Url/Factory/GitBlit.java

```diff
@@ -0,0 +1,132 @@
+package uk.co.ben_gibson.repositorymapper.Host.Url.Factory;
+
+import com.intellij.util.containers.hash.LinkedHashMap;
+import org.jetbrains.annotations.NotNull;
+import uk.co.ben_gibson.repositorymapper.Context.Context;
+import uk.co.ben_gibson.repositorymapper.Host.Url.Exception.ProjectNotFoundException;
+import uk.co.ben_gibson.repositorymapper.Repository.Exception.RemoteNotFoundException;
+
+import java.io.UnsupportedEncodingException;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Creates a URL in a format expected by GitBlit.
+ */
+public class GitBlit implements Factory {
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @NotNull
+    public URL createUrl(@NotNull Context context, boolean forceSSL) throws URISyntaxException, UnsupportedEncodingException, MalformedURLException, RemoteNotFoundException, ProjectNotFoundException {
+
+        String remoteUrl = context.getRepository().getOrigin().getFirstUrl();
+
+        /**
+         * Taken from https://github.com/markiewb/nb-git-open-in-external-repoviewer
+         * <pre>
+         gitblit.source.url.0=(?<protocol>http|https)://(?<username>.+?@)?(?<server>.+?)/(git/r)/(?<repo>.+)\.git
+         gitblit.source.url.1=(?<protocol>http|https)://(?<username>.+?@)?(?<server>.+?)/(git|r)/(?<repo>.+)\.git
+         gitblit.target.show_commitdiff.url.0=<protocol>://<server>/git/commitdiff/<repo|escapeSlashWithBang>.git/<revision>
+         gitblit.target.show_commitdiff.url.1=<protocol>://<server>/commitdiff/<repo|escapeSlashWithBang>.git/<revision>
+         gitblit.target.show_file.url.0=<protocol>://<server>/git/blob/<repo|escapeSlashWithBang>.git/<revision>/<fullfilepath|escapeSlashWithBang>#L<linenumber|1based>
+         gitblit.target.show_file.url.1=<protocol>://<server>/blob/<repo|escapeSlashWithBang>.git/<revision>/<fullfilepath|escapeSlashWithBang>#L<linenumber|1based>
+         * </pre>
+         */
+
+
+        //Note: Use LinkedHashMap because ordering the patterns have a priority
+        Map<String, String> commitDiffPatterns = new LinkedHashMap<>();
```

Comment:
Would it be more succinct to use an approach similar to the other factories over this patten mapping e.g

```
String path;
String fragment = null;

if (context.getCommitHash() != null) {
    path = String.format("%s/commitdiff/%s", remoteUrl.getPath(), context.getCommitHash());
} else {
    path = String.format(
        "%s/blob/%s%s",
        remoteUrl.getPath(),
        context.getBranch(),
        context.getFilePathRelativeToRepository()
    );

    if (context.getCaretLinePosition() != null) {
        fragment = "L" + context.getCaretLinePosition().toString();
    }
}
```